### ISSUE TITLE
bundler tries to trap SIGABRT

### ISSUE THREAD

**Issue by dreyks** [rubyomr-preview/ruby#24]
Anything that runs through `bundle exec` fails with the following error `ArgumentError: can't trap reserved signal: SIGABRT`

As seen in [here](https://github.com/bundler/bundler/blob/1f10c7f07fdbe5b664272d54657b5687bc881810/lib/bundler/cli/exec.rb#L8) bundler considers only `SEGV BUS ILL FPE VTALRM KILL STOP` as reserved signals and then it [traps](https://github.com/bundler/bundler/blob/1f10c7f07fdbe5b664272d54657b5687bc881810/lib/bundler/cli/exec.rb#L72-L73) all the signlas except those.

Additionaly, when I added `ABRT` to reserved signals it still continued to fail with the same error until I also added `IOT` to the same list. Only then it passed only to fail on nokogiri, but that is a thing for another issue

why is `SIGABRT` reserved in ruby-omr but not reserved in original ruby?

**Comment by dreyks** [comment]
ruby_2_4_omr_preliminary branch does not have this problem though

**Comment by mgaudet** [comment]
Interesting. (Thanks for checking the `ruby_2_4_omr_preliminary` branch too!) 

@charliegracie Any initial thoughts on this? I'll dig in this afternoon if nothing stands out to you.

**Comment by mgaudet** [comment]
It appears to have been done to support failing OMR GC assertions. I suspect it's possible to restore the old behaviour: [Here's an untested patch](https://gist.github.com/mgaudet/a247522dfd95ce4cd8a3cd37c5d9a185) that reverts the commit where we added this. However, I'm hesitant to commit before I understand more what the impact would be. 

(This must be relatively new checking in bundler -- I definitely have used ruby_2_2_omr w/ bundler before!)

**Comment by dreyks** [comment]
Yes, this was introduced in 1.13 bundler/bundler@93e11fc + bundler/bundler@69a0f91
